1091. Shortest Path in Binary Matrix/solution.py

```diff
@@ -4,23 +4,77 @@ class Solution:
     # cells containing a 0. Each cell is connected to the 8 cells around it
     # that either share an edge or a corner. If no path is found, return -1.
     def shortestPathBinaryMatrix(self, grid: List[List[int]]) -> int:
+        # Check that top left and bottom right cells contain 0s. Otherwise no clear
+        # path can exist, so return -1
+        gridHeight = len(grid)
+        gridWidth = len(grid[0])
+        
+        if grid[0][0] != 0 or grid[gridHeight-1][gridWidth-1] != 0:
+            return -1
+        
         # Use BFS to find the shortest path so keep track of nodes
         # to visit in a queue. Each entry is a tuple of the col and row index of
-        # the cell to visit.
-
-        # Use a grid of bool to track if nodes have been visited.
+        # the cell to visit. Initialize queue with top left cell.
+        queue = [(0,0)]
+        
+        # Set top left cell value to -1 to indicate it has been visited.
+        grid[0][0] = -1
 
         # Loop while there are more nodes to visit. Each loop here represents a round.
+        # Start roundNum at 1 since a grid of 1x1 will have a path length of 1.
+        roundNum = 1
+        while queue:
 
             # Go through each of the nodes currently in the queue which will be all
             # the nodes we process in this round
+            numCurrentRoundCells = len(queue)
+            while numCurrentRoundCells > 0:
+
+                # pop next cell from front of queue to process
+                (row, col) = queue.pop(0)
```

Comment:
I like your (row, col) tuple idea! I think you should explain strictly what that means. Although it might not matter for coders that don't get confused with row and column, it'll be helpful for your reader.

Reply:
It's explained in the comment initializing the queue.